app/decorators/swagger/route_decorator.rb

```diff
@@ -14,6 +15,20 @@ def to_swagger(api_gateway_integration)
     }.select { |_, v| v.present? }
   end
 
+  def swagger_operation_id
+    return operation_id if operation_id.present?
+
+    result = http_method.downcase
+    url.split('/').map do |part|
+      if part.starts_with?(':')
+        result += "By#{part[1..-1].capitalize}"
+      else
+        result += part.capitalize
+      end
+    end
+    result.delete("^a-zA-Z0-9")
```

Comment:
Example of results:

| Route                          | Operation id                |
|--------------------------------|-----------------------------|
| GET /reports                   | getReports                  |
| GET /reports/:id               | getReportsById              |
| GET /reports/:id/detail        | getReportsByIdDetail        |
| PUT /reports/:id/detail/:subid | putReportsByIdDetailBySubid |

It is *not* the better naming but give something readable, searchable and avoid name conflicts. It should be considered as a workaround to unnamed operation id.